From c9945cd40f86c7faaed6a1cdace641155d1304b4 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 9 May 2019 18:31:07 +0800
Subject: [PATCH 27/30] mxc: vpu: fix compile warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

./include/linux/kernel.h:852:39: warning: comparison between ‘enum <anonymous>’ and ‘enum <anonymous>’ [-Wenum-compare]
 #define max(x, y) __careful_cmp(x, y, >)
                                       ^
./include/linux/kernel.h:828:30: note: in definition of macro ‘__cmp’
 #define __cmp(x, y, op) ((x) op (y) ? (x) : (y))
                              ^~
./include/linux/kernel.h:852:19: note: in expansion of macro ‘__careful_cmp’
 #define max(x, y) __careful_cmp(x, y, >)
                   ^~~~~~~~~~~~~
drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c:3535:10: note: in expansion of macro ‘max’
  count = max(GTB_ENC_CMD_RESERVED, VID_API_ENC_EVENT_RESERVED);
          ^~~

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index b8cdb7772975..5a6b8c255ffb 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -3532,7 +3532,7 @@ static int show_cmd_event_infos(struct vpu_statistic *statistic,
 
 	num += scnprintf(buf + num, size - num, "command/event:\n");
 
-	count = max(GTB_ENC_CMD_RESERVED, VID_API_ENC_EVENT_RESERVED);
+	count = max((int)GTB_ENC_CMD_RESERVED, (int)VID_API_ENC_EVENT_RESERVED);
 	for (i = 0; i <= count; i++)
 		num += show_cmd_event(statistic, i, buf + num, size - num);
 
-- 
2.17.1

